client/ayon_blender/api/plugin_load.py

```diff
@@ -0,0 +1,212 @@
+    # Giving it a 'random' object from the collection seems to override
+    # everything contained in the loaded collection.
+    context.view_layer.objects.active = loaded_objects[0]
+
+    from .plugin import create_blender_context  # todo: move import
+    operator_context = create_blender_context(
+        active=loaded_objects[0],
+        selected=loaded_objects
+    )
+
+    # https://blender.stackexchange.com/questions/289245/how-to-make-a-blender-library-override-in-python  # noqa
+    # https://docs.blender.org/api/current/bpy.types.ID.html#bpy.types.ID.override_hierarchy_create  # noqa
+    if bpy.app.version[0] >= 4:
+        with bpy.context.temp_override(**operator_context):
+            loaded_collection.override_hierarchy_create(
+                scene, context.view_layer, do_fully_editable=True
+            )
+    # Pre 4.0 method:
+    else:
+        pass
+
+    scene.collection.children.unlink(loaded_collection)
+
+    local_collection = get_local_collection(
+        overridden_collections,
+        loaded_collection,
+    )
+    return local_collection
+
+
+def get_local_collection(
+    overridden_collections: list[bpy.types.Collection],
+    loaded_collection: bpy.types.Collection,
+) -> bpy.types.Collection:
+    """Get the local (overridden) collection.
+
+    To get it we check all collections with a library override and check if
+    they have the loaded collection as their reference. If a collection is
+    not in the provided (known) override collections, we assume it's the newly
+    created one.
+    """
+    local_collections: set[bpy.types.Collection] = set()
+    for collection in get_overridden_collections_from_reference_collection(
+        loaded_collection
+    ):
+        if collection not in overridden_collections:
+            local_collections.add(collection)
+    if len(local_collections) != 1:
+        raise RuntimeError("Could not find the overridden collection.")
+
+    return local_collections.pop()
+
+
+def get_overridden_collections_from_reference_collection(
+    reference_collection: bpy.types.Collection,
+) -> Generator[bpy.types.Collection, None, None]:
+    """Get collections that are overridden versions of the reference collection.
+
+    Yields:
+        All collections that have an override library and have the
+        `reference_collection` collection as reference.
+    """
+    for collection in bpy.data.collections:
+        if not collection.override_library:
+            continue
+        if collection.override_library.reference == reference_collection:
+            yield collection
+
+
+def load_collection(
+    filepath,
+    link=True,
+    lib_container_name = None
+) -> bpy.types.Collection:
+    """Load a collection to the scene."""
+    with bpy.data.libraries.load(filepath, link=link, relative=False) as (
+        data_from,
+        data_to,
+    ):
+        if lib_container_name is None:
+            lib_container_name = data_from.collections[0]
```

Comment:
This will fail if the scene has no collections to begin with... which _may_ be possible? Or is that _impossible_ in Blender files I wonder?

Also, this isn't necessarily what I meant. I meant to find the first collection that is NOT a child of any other collections inside the blender file, but maybe the first collection is already that in blender lingo.

I also wonder what happens if on `update` the same named collection does not exist anymore. We may need to test that?

Reply:
> I also wonder what happens if on update the same named collection does not exist anymore. We may need to test that?

For the updated function, it would still search if there is existing collection and used that one. But the collection has been removed, it would show the corrupted file icon on the linked collection. We might need to re-load the new collection instead if the collection has been deleted.
![image](https://github.com/user-attachments/assets/2d6daaed-0888-4b8e-bd87-4a3e90abe853)